pipeline/acquisition.py

```diff
@@ -185,29 +186,66 @@ class Voltage(dj.Part):
         voltage_time_stamps: longblob
         """
         
+    def make(self,key):
+        # this function implements the ingestion of raw extracellular data into the pipeline
+        return None
 
 @schema
 class Spike(dj.Imported):
     definition = """ 
-    -> Extracellular
+    -> ProbeInsertion
     unit_id : smallint
     ---
-    spike_times: longblob
-    spike_type: varchar(32) # e.g. wide width, narrow width spiking
+    -> reference.Probe.Channel
+    spike_times: longblob # time of each spike, with respect to the start of session 
+    unit_cell_type: varchar(32) # e.g. cell-type of this unit (e.g. wide width, narrow width spiking)
     unit_depth_x: float
     unit_depth_y: float
     unit_depth_z: float
+    spike_waveform: longblob # waveform(s) of each spike at each spike time (spike_time x waveform_timestamps)
     """
-
-    class SpikeWaveForm(dj.Part):
-        definition = """ 
-        -> master
-        -> reference.Probe.Channel
-        ---
-        spike_waveform: longblob
-        spike_time_stamps: longblob
-        """   
         
+    def make(self,key):
+        ############## Dataset #################
+        sess_data_dir = os.path.join('..','data','extracellular','datafiles')
+                
+        # Get the Session definition from the keys of this session
+        animal_id = key['subject_id']
+        date_of_experiment = key['session_time']
+        
+        # Search the files in filenames to find a match for "this" session (based on key)
+        sess_data_file = utilities.find_session_matched_nwbfile(sess_data_dir, animal_id, date_of_experiment)
+        if sess_data_file is None: 
+            return
```

Comment:
hm, if there is nothing to insert, then this should be somehow reflected in the `key_source`.  If you just return without inserting, then datajoint will keep attempting to populate this entry.